common/scala/src/main/scala/whisk/core/entity/WhiskEntity.scala

```diff
@@ -158,9 +156,16 @@ case class LimitedWhiskEntityPut(
     parameters: Option[Parameters] = None,
     annotations: Option[Parameters] = None) {
 
-    def isWithinSizeLimits: Boolean = {
-        exec.map(_.size).getOrElse(0 B) <= Exec.sizeLimit &&
-            parameters.map(_.size).getOrElse(0 B) <= Parameters.sizeLimit &&
-            annotations.map(_.size).getOrElse(0 B) <= Parameters.sizeLimit
+    def isWithinSizeLimits: Option[(ByteSize, ByteSize)] = {
+        exec.flatMap { e =>
+            val is = e.size
+            if (is <= Exec.sizeLimit) None else Some(is, Exec.sizeLimit)
+        } orElse parameters.flatMap { p =>
+            val is = p.size
+            if (is <= Parameters.sizeLimit) None else Some(is, Parameters.sizeLimit)
+        } orElse annotations.flatMap { a =>
+            val is = a.size
+            if (is <= Parameters.sizeLimit) None else Some(is, Parameters.sizeLimit)
+        }
```

Comment:
The return value of this could be confusing with regards to the error message returned to the user. If I paste a 40MB file and 2MB parameters I will get: Something like: You sent 2Megabytes but only 1 is allowed. Maybe we should add the `field` as well and put that into the error message?